### PR TITLE
chore: make request timeout configurable in NotionDBLoader

### PR DESCRIPTION
**Pull request by parkedwards**
I am also hitting some timeouts on the Notion DB fetching (~20 documents in the source notion db)

https://github.com/hwchase17/langchain/pull/2056#issuecomment-1530002441

^ this user makes a good point about making this underlying fetch logic extensible for retries/Idempotence, but this PR just makes the request timeout configurable

### CODE REVIEW

File: langchain/document_loaders/notiondb.py [hunk 1]
```diff
@@ -19,14 +19,20 @@ class NotionDBLoader(BaseLoader):
     Args:
         integration_token (str): Notion integration token.
         database_id (str): Notion database id.
+        request_timeout_sec (int): Timeout for Notion requests in seconds.
     """
 
-    def __init__(self, integration_token: str, database_id: str) -> None:
+    def __init__(self, integration_token: str, database_id: str, request_timeout_sec: int = None) -> None:
```
Comment (pranjaldoshi96) [inline langchain/document_loaders/notiondb.py:25]: type is Optional[int]

File: langchain/document_loaders/notiondb.py [hunk 2]
```diff
@@ -19,14 +19,20 @@ class NotionDBLoader(BaseLoader):
...
         """Initialize with parameters."""
         if not integration_token:
             raise ValueError("integration_token must be provided")
         if not database_id:
             raise ValueError("database_id must be provided")
+        if request_timeout_sec > 30:
+            raise ValueError("request timeout cannot exceed 30s")
+
+        if not request_timeout_sec:
+            request_timeout_sec = 10
```
Comment (pranjaldoshi96) [inline langchain/document_loaders/notiondb.py:35]: Why it is 10, usually timeouts are set to 10. And if you want to default value to 10 then set it on param.

File: langchain/document_loaders/notiondb.py (hunk 1 above)
Comment (parkedwards) [reply langchain/document_loaders/notiondb.py]: ah right, thank you. fixed

File: langchain/document_loaders/notiondb.py (hunk 2 above)
Comment (parkedwards) [reply langchain/document_loaders/notiondb.py]: got it, i set the default value in the function argument instead.  it's 10s here bc that was the hardcoded value before

**Review by pranjaldoshi96 — APPROVED** [review]

File: langchain/document_loaders/notiondb.py [hunk 3]
```diff
@@ -19,14 +19,17 @@ class NotionDBLoader(BaseLoader):
...
     """
 
-    def __init__(self, integration_token: str, database_id: str) -> None:
+    def __init__(self, integration_token: str, database_id: str, request_timeout_sec: Optional[int] = 10) -> None:
         """Initialize with parameters."""
         if not integration_token:
             raise ValueError("integration_token must be provided")
         if not database_id:
             raise ValueError("database_id must be provided")
+        if request_timeout_sec > 30:
```
Comment (dev2049) [inline langchain/document_loaders/notiondb.py:31]: where does 30 come from? do we need to enforce an upper bound?

Comment (parkedwards) [reply langchain/document_loaders/notiondb.py:31]: technically no, but i didn't want to leave this unbounded. open to suggestions or just removing this check altogether

Comment (dev2049) [reply langchain/document_loaders/notiondb.py:31]: think ok to remove check since it would only be large if explicitly configured

Comment (parkedwards) [reply langchain/document_loaders/notiondb.py]: np, done

**Comment by parkedwards** [comment]
checking on the failed lint job, sec

**Review by hwchase17 — APPROVED** [review]
lgtm!

**Pull request merged**